dotnet/src/SemanticKernel.Abstractions/IKernel.cs

```diff
@@ -87,6 +87,16 @@ public interface IKernel
     /// <param name="memory">Semantic memory instance</param>
     void RegisterMemory(ISemanticTextMemory memory);
 
+    /// <summary>
+    /// Run a single synchronous or asynchronous <see cref="ISKFunction"/>.
+    /// </summary>
+    /// <param name="skFunction">A Semantic Kernel function to run</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Result of the function composition</returns>
+    Task<SKContext> RunAsync(
+        ISKFunction skFunction,
+        CancellationToken cancellationToken); // TODO: add `= default` value after the `params` overloads are removed, or it will be ambiguous.
+
```

Comment:
What's the plan for the below overloads, e.g.
```C#
    Task<SKContext> RunAsync(
        string input,
        CancellationToken cancellationToken,
        params ISKFunction[] pipeline);
```
?  Should they be obsoleted / deleted?

Reply:
Yeah, we'll obsolete the "params" ones. They don't allow for any arg passing or anything. 
Current thinking: Kernel either runs a `Function` or a `Plan`.
...where a `Plan` has support for chaining, args, other complexity.